src/CreateGoalPanel4Mobile/index.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import classNames from 'classnames';
+import CategoryButtons from '../CategoryButtons';
+import CreateGoalActionItem from '../CreateGoalActionItem';
+import SectionHeaderGeneral from '../SectionHeaderGeneral';
+import Spacer from '../Spacer';
+import styles from './styles.css';
+import SectionFooterNew from '../SectionFooterNew';
+import Wrapper from '../Wrapper';
+const PropTypes = React.PropTypes;
+
+function CreateGoalPanel4Mobile(props) {
+  return (
+    <div className={styles.outerWrapper}>
+      <SectionHeaderGeneral
+        children={props.children}
+        className={styles.sectionHeaderGeneral}
+        rightIcon={"icon-close"}
+        title={props.title}
+        />
+      <Wrapper
+        className={classNames(styles.wrapper, {
+          [props.className]: props.className
+        })}
+        size={'standard'}
+        >
+        <CategoryButtons
+          activeCategory={props.activeCategory}
+          onCategoryClick={props.onCategoryClick}
+          />
+      </Wrapper>
+      <CreateGoalActionItem className={styles.actionItemComposite}
+        co2={props.co2}
+        icon={props.icon}
+        onClick={props.onClickGoalActionItem}
+        points={props.points}
+        title={props.titleGoal1}
+        />
+      <Spacer
+        className={styles.hr2}
+        hr={"true"}
+        size={"hr2"}
+        />
+      <CreateGoalActionItem className={styles.actionItemComposite}
+        co2={props.co2}
```

Comment:
These `CreateGoalActionItem`components would be passed in by the parent since the number and type of activity isn't fixed. You can create a new function that expects a type of object then go through that object to make the number of action items.